lib/perl/Genome/Disk/Detail/Allocation/Unarchiver.pm

```diff
@@ -177,4 +133,49 @@ sub _swap_shadow_allocation {
     return 1;
 }
 
+sub _do_unarchive_cmd {
+    my $class = shift;
+    my $id = shift;
+    my $cmd = shift;
```

Comment:
It would be better if this sub took an allocation object and a destination path.  Then you can pull out construction of the ```tar``` command from the other two places into this function.